vaadinfrontend/src/main/java/life/qbic/datamanager/views/projects/project/samples/registration/batch/SampleRegistrationSpreadsheet.java

```diff
@@ -398,40 +409,80 @@ public List<SamplesheetHeaderName> retrieveGenomics() {
         SamplesheetHeaderName.CUSTOMER_COMMENT);
   }
 
+  public Result<Void, InvalidSpreadsheetRow> areInputsValid() {
+
+    Set<String> concatenatedSampleIDs = new HashSet<>();
+    for (int rowId = 1; rowId < sampleRegistrationSheet.getLastRowNum(); rowId++) {
+      Row row = sampleRegistrationSheet.getRow(rowId);
+      List<String> mandatoryInputs = new ArrayList<>();
+      for (SamplesheetHeaderName name : SamplesheetHeaderName.values()) {
+        if(name.isMandatory) {
+          mandatoryInputs.add(SpreadsheetMethods.cellToStringOrNull(row.getCell(
+              header.indexOf(name))));
+        }
+      }
+      // break when cells in row are undefined
```

Comment:
why do you break in rows with undefined cells? Is the assumption that the first row containing one or more undefined cells is the last row that is provided?

Reply:
yes, or rather the row after that, that the user can't fill any more